File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.cpp
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        
            int temp=k;
            k-=arr[0]-1;
            if(k<=0){
                
                    return temp;
            }
            
            for(int i=1;i<arr.size();i++){
                
                 
                   
                    if(k-(arr[i]-arr[i-1]-1)<=0){
                        
                            return arr[i-1]+k;
                        
                    }
                       k-=(arr[i]-arr[i-1]-1);
                
            }
        
            return arr[arr.size()-1]+k;
            
        
        
            
                
        
    }
};

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.py
class Solution:
    
    def findKthPositive(self, arr: List[int], k: int) -> int:
    
            start=0
            end=len(arr)-1
            while(start<end):
                mid=(start+end)>>1
                val=arr[mid]
                if (val-mid)<=k:
                    start=mid+1
                else:
                    end=mid
            
           
            if arr[start]-start-1<k:
                cnt=(k-(arr[start]-start-1))
                print(cnt)
                return arr[start]+cnt
            
            print("here")
            va=arr[start-1]-start
            
            return arr[start-1]+k-va
            
            